test/EIP7702Proxy/isValidSignature.t.sol

```diff
@@ -199,3 +308,76 @@ contract RevertingImplementationTest is IsValidSignatureTestBase {
         return ERC1271_FAIL_VALUE;
     }
 }
+
+/**
+ * @dev Tests isValidSignature behavior when implementation returns ERC1271_MAGIC_VALUE with extra data
+ */
+contract ExtraDataTest is IsValidSignatureTestBase {
+    function test_mockReturnsExtraData() public {
```

Comment:
this test just makes sure our mock is actually behaving how we want it to and returning 32 bytes of `1271 MAGIC_VALUE | 28 bytes other stuff`

Reply:
thanks for including! curious why this needs to be a separate contract in the same file though?

Reply:
there could be ways to formulate these tests all in the same test contract, but there would be a lot of repetition. this test file uses a base contract to extract common setup and factor out some common test cases/infra (like the fact that all cases should succeed when a valid eoa sig is provided) while giving each mock its own space for setup since these `isValidSignature` tests hinge largely on what the implementation's `isValidSignature` is doing -- does it succeed? fail? revert? return weird extra data? -- and therefore use different mocks